src/main/java/com/redhat/rhjmc/containerjfr/net/web/handlers/RecordingOptionsGetHandler.java

```diff
@@ -0,0 +1,150 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ * #L%
+ */
+package com.redhat.rhjmc.containerjfr.net.web.handlers;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+
+import org.openjdk.jmc.common.unit.IConstrainedMap;
+import org.openjdk.jmc.common.unit.IOptionDescriptor;
+import org.openjdk.jmc.flightrecorder.configuration.recording.RecordingOptionsBuilder;
+
+import com.google.gson.Gson;
+
+import com.redhat.rhjmc.containerjfr.commands.internal.RecordingOptionsBuilderFactory;
+import com.redhat.rhjmc.containerjfr.net.AuthManager;
+import com.redhat.rhjmc.containerjfr.net.TargetConnectionManager;
+
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.RoutingContext;
+
+public class RecordingOptionsGetHandler extends AbstractAuthenticatedRequestHandler {
+
+    static final String PATH = "/api/v1/targets/:targetId/recordingOptions";
+    protected final TargetConnectionManager connectionManager;
+    private final RecordingOptionsBuilderFactory recordingOptionsBuilderFactory;
+    private final Gson gson;
+    private static final Pattern NUMBER_PATTERN = Pattern.compile("\\d+");
+
+    @Inject
+    RecordingOptionsGetHandler(
+            AuthManager auth,
+            TargetConnectionManager connectionManager,
+            RecordingOptionsBuilderFactory recordingOptionsBuilderFactory,
+            Gson gson) {
+        super(auth);
+        this.connectionManager = connectionManager;
+        this.recordingOptionsBuilderFactory = recordingOptionsBuilderFactory;
+        this.gson = gson;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.GET;
+    }
+
+    @Override
+    public String path() {
+        return PATH;
+    }
+
+    @Override
+    public boolean isAsync() {
+        return false;
+    }
+
+    @Override
+    void handleAuthenticated(RoutingContext ctx) throws Exception {
+        Map<String, String> optionMap =
+                connectionManager.executeConnectedTask(
+                        getConnectionDescriptorFromContext(ctx),
+                        connection -> {
+                            RecordingOptionsBuilder builder =
+                                    recordingOptionsBuilderFactory.create(connection.getService());
+                            IConstrainedMap<String> recordingOptions = builder.build();
+
+                            Map<String, IOptionDescriptor<?>> TargetRecordingOptions =
+                                    connection.getService().getAvailableRecordingOptions();
+
+                            Map<String, String> map = new HashMap<String, String>();
+
+                            if (recordingOptions.get("toDisk") != null) {
```

Comment:
Can these three if/elses be refactored into a loop over a set of keys? And rather than using constant strings like `toDisk`, could these use the RecordingOptionsCustomizer.OptionKeys?

Reply:
I made a similar loop in the PATCH handler, but this one makes less sense because toDisk has to be handled differently than the other two. It would have to be a loop for just two of the options.

Reply:
How about extracting utility methods for handling plain textual values like `toDisk` and numeric values like the other two?

Reply:
How would that work, how would the utility method know which type it need to check? woudl this still be in a loop?

Reply:
No loop. Make a utility method like `getNumericOption` and another like `getBooleanOption`.

Reply:
I created a helper function for the numeric ones, I don't know if it makes sense for the boolean one because for some reason the name of the option is different between recordingOptionsCustomizer and the target recordingOptions default